encryptor/postgresql/utils_test.go

```diff
@@ -0,0 +1,485 @@
+		}{
+			{
+				query: `SELECT "id", "email", "mobile_number" AS "mobileNumber" FROM "users" AS "User" where "User"."is_active"`,
+				expectedValues: []*base.ColumnInfo{
+					{Alias: "User", Table: "users", Name: "id"},
+					{Alias: "User", Table: "users", Name: "email"},
+					{Alias: "User", Table: "users", Name: "mobile_number"},
+				},
+			},
+			{
+				query: `SELECT "id", "email", "mobile_number" AS "mobileNumber" FROM "users" AS "User", "table1" as "test_table"`,
+				expectedValues: []*base.ColumnInfo{
+					{Alias: "User", Table: "users", Name: "id"},
+					{Alias: "User", Table: "users", Name: "email"},
+					{Alias: "User", Table: "users", Name: "mobile_number"},
+				},
+			},
+			{
+				query: `SELECT "id", "email", "mobile_number" AS "mobileNumber" FROM "users" AS "User", "users_duplicate" as "User2"`,
+				expectedValues: []*base.ColumnInfo{
+					nil, nil, nil,
+				},
+			},
+			{
+				query: `SELECT "id", "email", "mobile_number", "id_tmp", "email_tmp", "mobile_number_tmp"  AS "mobileNumber" FROM "users" AS "User", "users_temp" as "temp"`,
+				expectedValues: []*base.ColumnInfo{
+					{Alias: "User", Table: "users", Name: "id"},
+					{Alias: "User", Table: "users", Name: "email"},
+					{Alias: "User", Table: "users", Name: "mobile_number"},
+					{Alias: "temp", Table: "users_temp", Name: "id_tmp"},
+					{Alias: "temp", Table: "users_temp", Name: "email_tmp"},
+					{Alias: "temp", Table: "users_temp", Name: "mobile_number_tmp"},
+				},
+			},
+		}
+		for i, tcase := range testcases {
+			parsed, err := pg_query.Parse(tcase.query)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			columns, err := MapColumnsToAliases(parsed.Stmts[0].GetStmt().GetSelectStmt(), schemaStore)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if len(columns) != len(tcase.expectedValues) {
+				t.Fatal("Returned incorrect length of values")
+			}
+
+			for y, column := range columns {
+				if column == nil {
+					if tcase.expectedValues[y] != nil {
+						t.Fatalf("[%d] expected nil column value ", i)
+					}
+					continue
+				}
+
+				if *column != *tcase.expectedValues[y] {
+					t.Fatalf("[%d] Column info is not equal to expected - %+v, actual - %+v", i, tcase.expectedValues[i], *column)
+				}
+			}
+		}
+	})
+	t.Run("Join enumeration fields query", func(t *testing.T) {
+		queries := []string{
+			`select table1.number, from_number, to_number, type, amount, created_date
+				from table1 join table2 as t2 on from_number = t2.number or to_number = t2.number join users as u on t2.user_id = u.id`,
+
+			// select with revers order of JOINs declaration
+			`select t3.*, t1.*, t2.*
+				from table1 AS t1 join table2 as t2 on from_number = t2.number or to_number = t2.number join table3 as t3 on t2.user_id = t3.id`,
+
+			// case with multiple table JOIN block ParenTableExpr
+			`select t1.number AS t1_number, t2.number AS t2_number from (select * from tablex) AS t JOIN (table1 AS t1 JOIN table2 AS t2 ON t1.id = t2.exam_type_id) ON t.version_id =
```

Comment:
oh, this query broke my brain. I don't remember such syntax `JOIN (...)` without alias for join statement